Add TLS/mTLS support to TCP Router

Birdrock opened the pull request:
- [ ] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
Adds TLS/mTLS support to TCP Router.


Backward Compatibility
---------------
Breaking Change? **No**

Birdrock committed:
Add TLS/mTLS support to TCP Router

- Add TLS cert, private key, ca

Signed-off-by: Geoff Franks <geoff.franks@broadcom.com>

geofffranks reviewed (commented):
BOSH link support was not added for pulling in the CA, because gorouter does not use links to pull in the CA for its backend tls info, and we were mimicking its behavior here.

Additionally rep does not provide a link with its executor's instance identity CA.

ameowlia commented on jobs/tcp_router/templates/tcp_router.yml.erb:
```diff
@@ -58,3 +58,29 @@ routing_api:
 haproxy_pid_file: "/var/vcap/data/tcp_router/config/haproxy.pid"
 isolation_segments: <%= p("tcp_router.isolation_segments") %>
 reserved_system_component_ports: <%=  reserved_system_component_ports %>
+<%
```
Sometimes people accidentally put in spaces or new lines for certs when they don't actually want anything set. Esp when using GUIs to set these properties. 

❓ Can we add a space/newline trim to this logic? And if there were only spaces then it should treat it as if nothing was provided?

Birdrock replied:
We definitely can. The block is already trimming trailing white space, so we can trim leading, as well.

One thought is that it might be using a lack of trimming to provide a new line.

Birdrock replied:
This looks like an execution block. Is the request to trim `ca_cert`, `client_cert`, and `client_key` individually? Since the values are there to check for various permutations, but aren't part of the output, do we still want to strip leading white space?

geofffranks replied:
Correct, i think we just want to call `.strip` when setting those three variables.

Birdrock committed:
Remove whitespace from certs/keys in templates

- tcp_router_backend_tls_ca.cert
- tcp_router_backend_tls.client_cert
- tcp_router_backend_tls.client_key

geofffranks commented:
Closing as this code was included in #425 with a couple extra commits that addressed PR feedback here + @Birdrock is temporarily on another project and we can't push to his branch.
